### PR TITLE
Refactor/radar search

### PR DESCRIPTION
**Pull request by MaxProtos**

### CODE REVIEW

File: src-fe/src/app/libs/common-components/radar-search/radar-search.component.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+				  #radarSearchAutocomplete="matAutocomplete"
+				  (optionSelected)="selectAutoCompleteOption($event)"
+				  disableRipple="true">
+	<perfect-scrollbar>
+		<mat-option *ngFor="let option of autoCompleteOptions"
+					[value]="option">
+			<span class="radar-search__option-label"
+				  [class.radar-search__option-label--no-content]="!hasOptionContent(option)"
+				  [innerHTML]="option.label | highlightText : searchQuery"></span>
+			<small class="radar-search__option-content"
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-search/radar-search.component.html:29]: why not div here?

File: src-fe/src/app/libs/common-components/text-input/text-input.theme.scss [hunk 2]
```diff
@@ -33,33 +33,3 @@
 		}
 	}
 }
-
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/text-input/text-input.theme.scss:36]: can we remove this component from our codebase completely?

File: src-fe/src/app/libs/radar-view/radar-view.component.ts [hunk 3]
```diff
@@ -135,23 +137,49 @@ export class RadarViewComponent implements OnInit, OnDestroy {
 	}
 
 	private updateAutoCompleteOptions(items: RadarDataItem[], query: string): void {
+		if (!Boolean(query)) {
+			this.autoCompleteOptions = [];
```
[the remark this answers is not in the document]
Comment (clain-atomi) [reply src-fe/src/app/libs/radar-view/radar-view.component.ts:141]: Or even handle this in parent function, to keep this function do 1 thing

File: src-fe/src/app/libs/radar-view/radar-view.component.ts [hunk 4]
```diff
@@ -135,23 +137,49 @@ export class RadarViewComponent implements OnInit, OnDestroy {
...
 	}
 
+	private removeTagsFromContent(content: string): string {
+		const htmlContainer: HTMLElement = document.createElement('div');
+		htmlContainer.innerHTML = content;
+
+		return htmlContainer.textContent;
+	}
+
+	private cutOptionsContentFromQueryEntry(content: string, query: string): string {
```
Comment (clain-atomi) [inline src-fe/src/app/libs/radar-view/radar-view.component.ts:171]: generally lets move all logic related to autocomplete calculation to separate service, radar view becomes fat controller

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: src-fe/src/app/libs/common-components/radar-search/radar-search.component.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+			  (click)="resetSearchQuery()"
+			  size="xs"
+			  icon="close">
+	</app-icon>
+</app-input-text>
+<mat-autocomplete class="radar-search__autocomplete"
+				  [class]="isDarkThemeActive ? 'radar-search__autocomplete--dark' : ''"
+				  #radarSearchAutocomplete="matAutocomplete"
+				  (optionSelected)="selectAutoCompleteOption($event)"
+				  disableRipple="true">
```
[the remark this answers is not in the document]
Comment (MaxProtos) [reply src-fe/src/app/libs/common-components/radar-search/radar-search.component.html:22]: Untitled
github troubles

File: src-fe/src/app/libs/common-components/text-input/text-input.theme.scss (hunk 2 above)
Comment (MaxProtos) [reply src-fe/src/app/libs/common-components/text-input/text-input.theme.scss:36]: not yet. It uses in others components

File: src-fe/src/app/libs/common-components/radar-search/radar-search.component.html (hunk 5 above)
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-search/radar-search.component.html:22]: indentation

File: src-fe/src/app/libs/common-components/radar-search/radar-search.component.scss [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	color: $font-color-gray;
+}
+
+.radar-search__input--dark .radar-search__reset-button {
+	color: $panel-border-dark;
+
+	&:hover {
+		color: $color-white;
+	}
+}
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-search/radar-search.component.scss:41]: .radar-search-input-dark { .radar-search__search-icon { ... } ...}

File: src-fe/src/app/libs/common-components/radar-search/radar-search.component.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+	@Output()
+	public autoCompleteOptionSelected: EventEmitter<AutoCompleteOption> = new EventEmitter<AutoCompleteOption>();
+
+	public onAutoCompleteValueChange(searchQuery: string): void {
+		this.searchQuery = searchQuery;
+		this.searchQueryChange.emit(searchQuery);
+	}
+
+	public get isDarkThemeActive(): boolean {
+		return this.theme === 'dark';
```
Comment (clain-atomi) [inline src-fe/src/app/libs/common-components/radar-search/radar-search.component.ts:33]: this.theme === ComponentTheme.Dark

**Review by clain-atomi — CHANGES_REQUESTED** [review]

**Review by clain-atomi — APPROVED** [review]

**Pull request merged**